python/rmm/_lib/device_buffer.pyx

```diff
@@ -278,17 +278,35 @@ cdef class DeviceBuffer:
 
         return b
 
-    cdef size_t c_size(self):
-        return self.c_obj.get()[0].size()
+    cdef size_t c_size(self) except *:
+        if self.c_obj:
+            return self.c_obj.get()[0].size()
+        else:
+            return 0
```

Comment:
After handing ownership of the data of a device buffer from Python back to C++, should it be UB/segfault to call any Python methods such as `DeviceBuffer.size`? Should we return 0/None? Or should we raise? @jakirkham @kkraus14 

In any case, would it be better to leave `c_size` as is, and handle the check for `self.c_obj` in the Python method `size` instead?

Reply:
fwiw, I think a "released-from" buffer should be similar to a "moved-from" buffer, i.e., it's the same as an empty buffer.

Reply:
In fact, if you just simply construct the returned `unique_ptr<device_buffer>` by move constructing from the wrapped `device_buffer`, then you shouldn't need to do anything special casing and you'll just rely on the behavior of a moved-from `device_buffer` being the same as an empty buffer.

Reply:
Having segfaults in Python sounds dangerous. That could come up anywhere and have unexpected affects.

I think we should just fill in `c_obj` to have an empty `device_buffer` after releasing. This should make sure other methods behave ok with sensible defaults.

Reply:
Thanks @jrhemstad - that worked!